Sources/SwerverCore/Response/ResponseBuilder.swift

```diff
@@ -2,35 +2,29 @@ import Foundation
 
 public class ResponseBuilder {
     private(set) var statusCode: StatusCode?
-    private(set) var contentType: ContentType?
     private(set) var body: String = String()
-    private(set) var contentLength: Int = 0
-    private(set) var allowedMethods: [String] = []
+    private(set) var headers: [ResponseHeader:String] = [:]
 
-    public init() {}
+    public init() {
+}
 
     public func withStatusCode(statusCode: StatusCode) -> ResponseBuilder {
         self.statusCode = statusCode
         return self
     }
-
-    public func withContentType(contentType: ContentType) -> ResponseBuilder {
-        self.contentType = contentType
-        return self
-    }
-
+    
     public func withBody(body: String) -> ResponseBuilder {
         self.body = body
-        return self
+        return setHeader(header: .contentLength, value: String(body.count))
     }
 
-    public func withContentLength(contentLength: Int) -> ResponseBuilder {
-        self.contentLength = contentLength
+    public func withContentType(contentType: ContentType) -> ResponseBuilder {
+        headers[.contentType] = contentType.rawValue
         return self
     }
     
-    public func withAllowedMethods(allowedMethods: [String]) -> ResponseBuilder {
-        self.allowedMethods = allowedMethods.sorted(by: <)
+    public func setHeader(header: ResponseHeader, value: String) -> ResponseBuilder {
```

Comment:
`setHeader(header:)` is redundant. It can just be `set(header:)`

This can be done as part of a future commit/PR.

Reply:
noted in new trello card